syntax = "proto3";

package Person_proto;

import "google/protobuf/empty.proto";


service PersonModelController{

    rpc List(PersonModelListRequest) returns (stream PersonModel){}
    rpc Create(PersonModel) returns (PersonModel){}
    rpc Retrieve(PersonModelRetrieveRequest) returns (PersonModel){}
    rpc Update(PersonModel) returns (PersonModel) {}
    rpc Destroy(PersonModel) returns (google.protobuf.Empty) {}
}

message PersonModel{
    
    int32 id = 1;
    
    int32 userId = 2;

    string firstname = 3;

    string lastName = 4;

    string username = 5;

    string password = 6;

    string email = 7;
}

message PersonModelListRequest{

}

message PersonModelRetrieveRequest{
    int32 id = 1;
}